From 1807794ea581c3a170cb773b3062133c4656bd8e Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 17 Feb 2021 19:12:08 +0200
Subject: [PATCH] [SW-20154] change flash valid address mask

Flash valid address changes to 2MB for Gaudi2,
whereas 16MB for Greco. Hence move it to platform
specific code

Change-Id: I7627c9f966af29d6f654ee8de5337c4c76272a70
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/spi_flash.h b/common/spi_flash.h
index 52491a5..9f10a09 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -25,7 +25,6 @@
 
 #define SPI_FLASH_BAUDR_DIVISOR		2
 
-#define FLASH_VALID_MASK		0xFFFFFF
 #define FLASH_64KB_PADD			65536
 #define FLASH_FW_VERSION_SIZE		16
 #define FLASH_KEY_HASH_SIZE		160
diff --git a/gaudi2_arc_gnu/soc.h b/gaudi2_arc_gnu/soc.h
index 53a5172..1778b5f 100644
--- a/gaudi2_arc_gnu/soc.h
+++ b/gaudi2_arc_gnu/soc.h
@@ -17,6 +17,9 @@
 
 #define SPI_FLASH_BLOCK_SIZE			1024
 
+/* 2MB for Gaudi2 */
+#define FLASH_VALID_MASK			0x1FFFFF
+
 #define HL_SRAM_BASE				0x1000007FF0000000ULL
 #define HL_SP_SRAM_BASE            		0x1000007FF7FE0000ULL
 
diff --git a/goya2_arc/soc.h b/goya2_arc/soc.h
index 6ebd1a5..55aac0e 100644
--- a/goya2_arc/soc.h
+++ b/goya2_arc/soc.h
@@ -21,6 +21,9 @@
 
 #define SPI_FLASH_BLOCK_SIZE			1024
 
+/* 16MB for Greco */
+#define FLASH_VALID_MASK			0xFFFFFF
+
 #define HL_SRAM_BASE				0x7FF0000000ULL
 #define HL_SP_SRAM_BASE            		0x7FFBFE0000ULL
 #define HL_SDRAM_BASE				0x10000000
